### PR TITLE
Intern Task Alina Mulik: Homework

### PR DESCRIPTION
**Pull request by alina-mulik**
Added homework 9

### CODE REVIEW

File: LinkedListClass/LinkedList.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+﻿namespace LinkedListClass
+{
+    public class LinkedList
```
Comment (yauhenists) [inline LinkedListClass/LinkedList.cs]: You don't need to create a new custom class, use the LinkedList<> from System.Collections

File: QueueAndStack/Program.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
+﻿namespace QueueAndStack
+{
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            // Task 1:
+            var newQueue = AddValuesOfTheQueue();
+            Console.WriteLine($"The queue with the deleted element {DeleteValueInTheQueue(newQueue)}");
```
Comment (yauhenists) [inline QueueAndStack/Program.cs]: The output in the console is the type of the object

File: QueueAndStack/Program.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                catch (FormatException)
+                {
+                    Console.WriteLine("You've entered invalid value!");
+                }
+            }
+        }
+
+        public static void SortStackDesc(Stack<char> stack)
+        {
+            var orderedList = stack.OrderByDescending(x => x).ToList();
```
Comment (yauhenists) [inline QueueAndStack/Program.cs]: The task is to write in console in reverse ordering of the input order. E.g. you've entered: q, a, z. Expected result - z, a, q

File: DictionaryHomework9/Program.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+         * На этот раз используйте другое имя и другой возраст. Наконец, прочитайте первый элемент, который вы добавили
+         * в словарь, и запишите его в консоль с помощью Console.WriteLine.*/
+        public static void AddValuesToDictAndOutputThem()
+        {
+            Dictionary<string, int> dictionary = new Dictionary<string, int>();
+            dictionary.Add("Alina", 26);
+            dictionary["Kate"] = 29;
+            try
+            {
+                var search = dictionary.Where(p => p.Key.Contains("Alina"));
```
Comment (yauhenists) [inline DictionaryHomework9/Program.cs]: I think in the task it means the first element of the collection without any filtering

**Review by yauhenists — CHANGES_REQUESTED** [review]

File: QueueAndStack/Program.cs (hunk 2 above)
Comment (alina-mulik) [reply QueueAndStack/Program.cs]: 😞 Fixed.

File: QueueAndStack/Program.cs (hunk 3 above)
Comment (alina-mulik) [reply QueueAndStack/Program.cs]: 😞 Fixed.

File: DictionaryHomework9/Program.cs (hunk 4 above)
Comment (alina-mulik) [reply DictionaryHomework9/Program.cs]: 😞 Fixed.

File: LinkedListClass/LinkedList.cs (hunk 1 above)
Comment (alina-mulik) [reply LinkedListClass/LinkedList.cs]: This one blew up my brain 🤯 Hopefully, fixed.

File: LinkedListHomework9/Program.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        /* Task 1. Создайте LinkedList и два элемента, вставьте второй элемент
+         * после каждого вхождения первого элемента в списке. https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.linkedlist-1?view=net-6.0
+         * Итак, если список равен [2,4,3,2,8,2,5,1,2], а элементы равны 2 и 10,
+         * результатом будет [2,10,4,3,2,10,8,2,10,5,1,2,10] */
+        private static void InsertNodesAfterSelectedValue(LinkedList<int> linkedList, int firstElement, int secondElement)
+        {
+            var changedList = linkedList.ToList();
+            LinkedList<int>.Enumerator em = linkedList.GetEnumerator();
+            var indexHelper = 0;
+            while (em.MoveNext())
```
Comment (yauhenists) [inline LinkedListHomework9/Program.cs:37]: JFYI - Another solution for LinkedList:

            var currentNode = linkedList.First;

            while (currentNode != null)
            {
                if (currentNode.Value == firstElement)
                {
                    linkedList.AddAfter(currentNode, new LinkedListNode<int>(secondElement));
                    currentNode = currentNode.Next;
                }

                currentNode = currentNode.Next;
            }

**Review by yauhenists — APPROVED** [review]

File: LinkedListHomework9/Program.cs (hunk 5 above)
Comment (yauhenists) [reply LinkedListHomework9/Program.cs:37]: Just close this comment and merge

Comment (alina-mulik) [reply LinkedListHomework9/Program.cs:37]: Much better!  🤔  Thank you!

**Pull request merged**